File: src/main/java/org/chousik/collection/builder/collectors/DoubleCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class DoubleCollector extends NumberCollector<Double>{

    @Override
    public Double ask(String name, IValidator<Double> validator) throws ScriptExecutionError {
        return askNumber(name, validator, Double::parseDouble);
    }
}

File: src/main/kotlin/collection/builder/collectors/DoubleCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.validators.IValidator
import org.chousik.exception.ScriptExecutionError

class DoubleCollector : NumberCollector<Double?>() {

    override fun ask(name: String?, validator: IValidator<Double?>?): Double? {
        return askNumber(name!!, validator!!, java.lang.Double::parseDouble)
    }
}